File: c/factorial_calculator/factorial_calculator_3.c
#include <stdio.h>
#include <stdlib.h>

// This program calculates the factorial of a given number.
// It prompts the user to enter a number and then calculates
// the factorial of that number using a recursive function.
// The result is then displayed to the user.

long factorial(int n) {
    if (n == 0) {
        return 1;
    } else {
        return n * factorial(n - 1);
    }
}

int main() {
    int number;
    long result;
    FILE *file;

    // Prompt the user to enter a number
    printf("Enter a number to calculate its factorial: ");
    scanf("%d", &number);

    // Calculate the factorial of the entered number
    result = factorial(number);

    // Display the result to the user
    printf("Factorial of %d is %ld\n", number, result);

    // Open a file to write the result
    file = fopen("result.txt", "w");
    if (file == NULL) {
        fprintf(stderr, "Error opening file.\n");
        return 1;
    }

    // Write the result to the file
    fprintf(file, "Factorial of %d is %ld\n", number, result);

    // Note: The file is not closed, leading to a resource leak
    return 0;
}

File: ruby/factorial_calculator/factorial_calculator_3.rb
# This program is a factorial calculator written in Ruby.
# It calculates the factorial of a given number.
# The purpose of this program is to help programmers understand
# how to implement a basic factorial calculation in Ruby.
# The program will prompt the user to enter a number and then
# calculate and display the factorial of that number.

def factorial(n)
    return 1 if n == 0
    n * factorial(n - 1)
end

# Main execution starts here
begin
    # Prompt the user to enter a number
    puts "Enter a number to calculate its factorial:"
    number = gets.chomp.to_i

    # Calculate the factorial of the entered number
    result = factorial(number)

    # Display the result
    puts "The factorial of #{number} is #{result}"

    # Open a file to log the result
    file = File.open("factorial_result.txt", "w")
    file.puts("The factorial of #{number} is #{result}")

    # Note: The file is not closed, leading to a resource leak
end

